doc/namespace-store-crd.md

```diff
@@ -1,157 +1,173 @@
 [NooBaa Operator](../README.md) /
 # NamespaceStore CRD
+The NamespaceStore CRD represents a storage target to be used as underlying storage for the data in NooBaa buckets (i.e. the buckets that are created by using the product)
+These storage targets are used to store and read plain data.
+NamespaceStores are referred to by name when defining [a BucketClass](bucket-class-crd.md).
 
-NamespaceStore CRD represents an underlying storage to be used as read or write target for the data in NooBaa 
-namespace buckets.
-These storage targets are used to store plain data.
-Namespace-Store are referred to by name when defining [BucketClass](bucket-class-crd.md).
-Multiple types of Namespace-Store are currently supported: aws-s3, s3-compatible, ibm-cos, google-cloud-storage, azure-blob.
+Supported NamespaceStore types:
+- aws-s3
+- s3-compatible
+- ibm-cos
+- google-cloud-storage
+- azure-blob
 
 # Definitions
-
 - CRD: [noobaa.io_NamespaceStores_crd.yaml](../deploy/crds/noobaa.io_namespacestores_crd.yaml)
 - CR: [noobaa.io_v1alpha1_NamespaceStore_cr.yaml](../deploy/crds/noobaa.io_v1alpha1_namespacestore_cr.yaml)
 
 
-# Reconcile
+# Reconciliation
+It is possible to create namespace stores by using the NooBaa CLI tool (from here on referred to in shell commands as `noobaa`), or by applying YAMLs.
+
+From here on out, fields that depend on the user's choice and input will be marked with `<>`. In cases where the value might not be inferrable from context, an additional explanation might be included.
+Note that the commands will apply to the namespace that's currently active (can be checked with `kubectl config current-context`). A command-specific namespace can be set by adding the `-n <NAMESPACE>` flag.
+Also note that, if possible and applicable, omitting the `access-key` and `secret-key` flags is recommended in order to avoid leakage of secrets. When both flags are ommitted, the CLI will prompt the user to enter the keys interactively.
 
-#### AWS-S3 type
+If you choose to use the CLI's `--secret-name` option, or to apply a YAML, please see [Store Secret Creation](store-connection-secrets.md).
 
-Create a cloud resource within the NooBaa brain and use S3 API for reading or writing data in the AWS cloud.
+## AWS S3
+Uses the S3 API for IO operations on plain data in AWS buckets
 ```shell
-noobaa -n noobaa namespacestore create aws-s3 bs --access-key KEY --secret-key SECRET --target-bucket BUCKET
+noobaa namespacestore create aws-s3 <NAMESPACESTORE NAME> --access-key <> --secret-key <> --target-bucket <>
 ```
 ```yaml
 apiVersion: noobaa.io/v1alpha1
 kind: NamespaceStore
 metadata:
   finalizers:
   - noobaa.io/finalizer
-  labels:
-    app: noobaa
-  name: bs
-  namespace: noobaa
+  name: <>
+  namespace: <>
 spec:
   awsS3:
     secret:
-      name: namespace-store-aws-s3-bs
-      namespace: noobaa
-    targetBucket: BUCKET
+      name: <>
+      namespace: <>
+    targetBucket: <>
   type: aws-s3
 ```
 
```

Comment:
I think you can add here `AWS-S3 Security Token Service (STS)` as you did in #1064.